include/golioth_sys.h

```diff
@@ -45,7 +45,7 @@ typedef struct {
     void* user_arg;
 } golioth_sys_timer_config_t;
 
-golioth_sys_timer_t golioth_sys_timer_create(golioth_sys_timer_config_t config);
+golioth_sys_timer_t golioth_sys_timer_create(golioth_sys_timer_config_t *config);
```

Comment:
How about adding `const` to this argument type? Since this is a config, it probably should be mutable only in context of application and not modified by the SDK itself.

Reply:
Good thought! I've updated